src/fn/watch-project.js

```diff
@@ -0,0 +1,125 @@
+const { fs } = require('../lib/sync-fs');
+const environment = require('../lib/environment');
+const process = require('process');
+const path = require('path');
+const { warn, error, info } = require('../lib/log');
+
+const convertForms = require('../lib/convert-forms');
+const uploadForms = require('../lib/upload-forms');
+const { convertContactForm } = require('./convert-contact-forms');
+const { uploadAppSettings } = require('./upload-app-settings');
+const { compileAppSettings } = require('./compile-app-settings');
+const { uploadCustomTranslations } = require('./upload-custom-translations');
+const { uploadResources } = require('./upload-resources');
+
+const formXLSRegex = /^[a-zA-Z_]*\.xlsx$/;
+const formXMLRegex = /^[a-zA-Z_]*\.xml$/;
+
+const debounceDelay = 150;
+
+function waitForSignal() {
+    return new Promise((resolve) => {
+        process.on('SIGINT', function () {
+            resolve();
+        });
+    });
+}
+
+let changeListenerWait = false;
+const changeListener = function (projectPath, api) {
+    return async (_, fileName) => {
+        if (changeListenerWait) return;
+        changeListenerWait = setTimeout(() => { changeListenerWait = false; }, debounceDelay);
+
+        if (fileName === 'app_settings.json') {
+            // ignore
+        } else if (fileName === 'resources.json' || fileName.match(/.*\.(png|svg)/)) {
+            await uploadResources(projectPath);
+        } else if (fileName.match(/messages-[\w]*\.properties/)) {
+            await uploadCustomTranslations(environment.apiUrl, projectPath, environment.skipTranslationCheck);
+        } else if (fileName.match(/.*\.js$/) || fileName.match(/.*\.json$/)) {
+            await compileAppSettings(projectPath);
+            await uploadAppSettings(api, projectPath);
+        } else {
+            warn('main listen: don\'t know what to do with', fileName);
+        }
+    };
+};
+
+let appFormListenerWait = false;
+const appFormListener = async (_, fileName) => {
+    if (appFormListenerWait) return;
+    appFormListenerWait = setTimeout(() => { appFormListenerWait = false; }, debounceDelay);
+
+    if (fileName.match(formXLSRegex)) {
```

Comment:
we should also listen to file changes in the <form-name>-media directories and upload the corresponding form